geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/pubsub/PubSubImpl.java

```diff
@@ -58,23 +62,25 @@ public int getSubscriptionCount() {
 
   @Override
   public long publish(RegionProvider regionProvider, byte[] channel, byte[] message) {
-    Set<DistributedMember> membersWithDataRegion = regionProvider.getRegionMembers();
-    @SuppressWarnings("unchecked")
-    ResultCollector<String[], List<Long>> subscriberCountCollector = FunctionService
-        .onMembers(membersWithDataRegion)
-        .setArguments(new Object[] {channel, message})
-        .execute(REDIS_PUB_SUB_FUNCTION_ID);
+    executor.submit(() -> internalPublish(regionProvider, channel, message));
 
-    List<Long> subscriberCounts = null;
+    return subscriptions.findSubscriptions(channel).size();
```

Comment:
consider calling findSubscriptions(channel).size() first and saving its value in "long result".
Then only call executor.submit if result > 0.
Then return result.

Reply:
`findSubscriptions` only considers local subscribers and doesn't account for remote ones. So, unfortunately, we still need to always issue the fn call.